src/contracts/interfaces/IReleaseManager.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "../libraries/OperatorSetLib.sol";
+
+interface IReleaseManagerErrors {
+    /// @notice Thrown when a release with the same digest already exists.
+    error ReleaseAlreadyExists();
+    /// @notice Thrown when a release with the same digest is already deprecated.
+    error ReleaseAlreadyDeprecated();
+}
+
+interface IReleaseManagerTypes {
+    /// @notice A semantic version.
+    /// @param major The major version.
+    /// @param minor The minor version.
+    /// @param patch The patch version.
+    struct Version {
+        uint16 major;
+        uint16 minor;
+        uint16 patch;
+    }
+}
+
+interface IReleaseManagerEvents is IReleaseManagerTypes {
+    /// @notice Emitted when a new release is published for an operator set.
+    /// @param operatorSet The operator set that the release was published for.
+    /// @param releaseDigest The unique identifier for the release content.
+    /// @param registryUrl The URL where the release artifacts can be found.
+    /// @param version The semantic version of the release.
+    event ReleasePublished(OperatorSet operatorSet, bytes32 releaseDigest, string registryUrl, Version version);
+
+    /// @notice Emitted when a release deprecation is scheduled for an operator set.
+    /// @param operatorSet The operator set that the release was deprecated for.
+    /// @param releaseDigest The unique identifier for the release content.
+    /// @param deprecationTimestamp The timestamp when the release will be deprecated.
+    event DeprecationScheduled(OperatorSet operatorSet, bytes32 releaseDigest, uint32 deprecationTimestamp);
+}
+
+interface IReleaseManager is IReleaseManagerErrors, IReleaseManagerEvents {
+    // WRITE
+
+    /// @notice Publishes a new release for an operator set.
+    /// @param operatorSet The operator set to publish the release for.
+    /// @param digest The unique identifier for the release content.
+    /// @param registryUrl The URL where the release artifacts can be found.
+    /// @param version The semantic version of the release.
+    /// @dev Only callable by addresses with permissioned by the operator set's AVS.
+    function publishRelease(
+        OperatorSet calldata operatorSet,
+        bytes32 digest,
+        string calldata registryUrl,
+        Version calldata version
```

Comment:
Can we make this into a struct called `Release`? And it should be an an array of `Release`.

Every release should have an array of digests associated with it.

Reply:
Or maybe `Release` just has `digest` and `registryUrl`. And the version stays the same across both.

Reply:
```
Release {
  bytest digest,
  string registryUrl,
  Version version
}
```